### ISSUE TITLE
Hot Reload: Making any change in Shared/NavMenu.razor does not hot reload, but rebuilds and refreshes the browser

### ISSUE THREAD

**Issue by MacdonaldRobinson** [dotnet/aspnetcore#34477]
### Describe the bug
Hot Reload: Making any change in Shared/NavMenu.razor does not hot reload, but rebuilds and refreshes the browser.

![image](https://user-images.githubusercontent.com/18366446/126069359-069f8928-8491-43bf-91f1-d4436f1c7001.png)

### To Reproduce
- In WSL - Ubuntu 20.04 LTS
- Create a new web app using "dotnet new blazorwasm"
- Use "code ." to open it in VS Code
- In the Terminal type "dotnet watch"
- Open the link in the browser
- Open Shared/NavMenu.razor and make any change
- In the console you will see the change IS detected however you get the error: "Unable to handle changes to /home/mac/src/blazorwasm/Shared/NavMenu.razor." the app then restarts and refreshes the browser

### Expectation
Any changes made in the file Shared/NavMenu.razor should be hot reloaded and not trigger a rebuild

### Output of `dotnet --info`
.NET SDK (reflecting any global.json):
 Version:   6.0.100-preview.6.21355.2
 Commit:    7f8e0d76c0

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /home/mac/dotnet/sdk/6.0.100-preview.6.21355.2/

Host (useful for support):
  Version: 6.0.0-preview.6.21352.12
  Commit:  770d630b28

.NET SDKs installed:
  6.0.100-preview.6.21355.2 [/home/mac/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0-preview.6.21355.2 [/home/mac/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0-preview.6.21352.12 [/home/mac/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

**Comment by dominiclapointe** [comment]
I do have the exact same behavior. Any code change (even in the html of a .razor page) give me this =>
![image](https://user-images.githubusercontent.com/47004797/127200894-123416e4-b505-4625-bf87-bad4da52a4c5.png)

I tried running dotnet watch inside the project folder, dotnet watch --project pachtocsproj run inside the solution folder. Any changes give me that.

```
PS C:\Users\dlapointe.BCAMERICAS\lean\UI> dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.100-preview.5.21302.13
 Commit:    d6380bcae7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-preview.5.21302.13\

Host (useful for support):
  Version: 6.0.0-preview.5.21301.5
  Commit:  ec3e0b276b
```

**Comment by Saladressing** [comment]
I got the same issue when trying to upgrade to sdk: `6.0.0-preview.6.21352.12`.
Downgrading back to sdk `6.0.100-preview.5.21302.13` resolves the `Hot reload of changes succeeded.` on `File changed: ...`.

```
.NET SDK (reflecting any global.json):
 Version:   6.0.100-preview.5.21302.13
 Commit:    d6380bcae7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-preview.5.21302.13\

Host (useful for support):
  Version: 6.0.0-preview.5.21301.5
  Commit:  ec3e0b276b
```

**Comment by pranavkm** [comment]
Thanks for the issue report. We've addressed this scenario to an extent with the 6.0-rc1 builds. 

* Adding a new NavMenu item would result in a rude edit (it produces a new lambda and that's currently a rude edit)
* Updating an existing NavMenu should hot reload as expected.

Please give the rc1 builds from https://github.com/dotnet/installer#installers-and-binaries a try and let us know if you continue seeing additional problems.
